src/community/backup-restore/core/src/main/java/org/geoserver/backuprestore/processor/CatalogItemProcessor.java

```diff
@@ -0,0 +1,518 @@
+        return bestEffort;
+    }
+    
+    @Override
+    public T process(T resource) throws Exception {
+
+        if (resource != null) {
+            if (isNew) {
+                // Disabling additional validators
+                ((CatalogImpl) this.catalog).setExtendedValidation(false);
+            }
+            
+            LOGGER.info("Processing resource: " + resource + 
+                    " - Progress: [" + currentJobExecution.getExecutedSteps() + "/" + currentJobExecution.getTotalNumberOfSteps() + "]");
+
+            if (resource instanceof WorkspaceInfo) {
+                if (!validateWorkspace((WorkspaceInfo) resource, isNew)) {
+                    LOGGER.warning("Skipped invalid resource: " + resource);
+                    
+                    logValidationExceptions(resource);
+                    
+                    return null;
+                }
+            } else if (resource instanceof DataStoreInfo) {
+                if (!validateDataStore((DataStoreInfo) resource, isNew)) {
+                    LOGGER.warning("Skipped invalid resource: " + resource);
+                    
+                    logValidationExceptions(resource);
+                    
+                    return null;
+                }
+                
+                WorkspaceInfo ws = ((DataStoreInfo) resource).getWorkspace();
+                if (this.catalog.getDefaultDataStore(ws) == null) {
+                    this.catalog.setDefaultDataStore(ws, (DataStoreInfo) resource);
+                }
+                
+            } else if (resource instanceof CoverageStoreInfo) {
+                if (!validateCoverageStore((CoverageStoreInfo) resource, isNew)) {
+                    LOGGER.warning("Skipped invalid resource: " + resource);
+                    
+                    logValidationExceptions(resource);
+                    
+                    return null;
+                }
+            }
+            else if (resource instanceof ResourceInfo) {
+                if (!validateResource((ResourceInfo) resource, isNew)) {
+                    LOGGER.warning("Skipped invalid resource: " + resource);
+                    
+                    logValidationExceptions(resource);
+                    
+                    return null;
+                }
+            }
+            else if (resource instanceof LayerInfo) {
+                ValidationResult result = null;
+                try {
+                    result = this.catalog.validate((LayerInfo) resource, isNew);
+                    if (!result.isValid()) {
+                        
+                        logValidationExceptions(resource);
+                        
+                        return null;
+                    }
+                } catch (Exception e) {
+                    LOGGER.warning("Could not validate the resource " + resource + " due to the following issue: " + e.getLocalizedMessage());
+                    
+                    if(!bestEffort) {
```

Comment:
Some best effort handling here, but not above. Btw, these bits of code are cpy pasted, maybe roll a method to delegate to instead